x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -268,6 +275,73 @@ public void testHighWatermarkCanBeExceededOnColdOrFrozenNode() throws Exception
         });
     }
 
+    public void testOvercommitInitializingSearchableSnapshotShards() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startNode(
+            Settings.builder()
+                .putList(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_HOT_NODE_ROLE.roleName())
+                .build()
+        );
+
+        InternalClusterInfoService masterInfoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(
+            ClusterInfoService.class
+        );
+        ClusterInfoServiceUtils.refresh(masterInfoService);
+
+        int nbIndices = createIndices();
+
+        assertAcked(
+            client().admin()
+                .cluster()
+                .preparePutRepository("repository")
+                .setType("mock")
+                .setSettings(Settings.builder().put("location", randomRepoPath()).build())
+        );
+        var mockRepository = (MockRepository) internalCluster().getCurrentMasterNodeInstance(RepositoriesService.class)
+            .repository("repository");
+        // Prevent searchable snapshot shards from quickly jumping from INITIALIZED to STARTED
+        mockRepository.setBlockOnceOnReadSnapshotInfoIfAlreadyBlocked();
+
+        createSnapshot("repository", "snapshot", nbIndices);
+
+        Map<String, Long> indicesStoresSizes = sizeOfShardsStores("index-*");
+        assertAcked(client().admin().indices().prepareDelete("index-*"));
+
+        String otherDataNode = internalCluster().startNode(
+            Settings.builder().put(NodeRoleSettings.NODE_ROLES_SETTING.getKey(), DiscoveryNodeRole.DATA_COLD_NODE_ROLE.roleName()).build()
+        );
+        ensureStableCluster(3);
+
+        String otherDataNodeId = internalCluster().getInstance(NodeEnvironment.class, otherDataNode).nodeId();
```

Comment:
nit: I think it's better to rely on the cluster state for this:
```
var clusterState = client().admin().cluster().prepareState().get().getState();
var nodeId = clusterState.nodes().resolveNode(otherNadaNode).getId();
```

Reply:
Done!